service-proxy.js

```diff
@@ -494,60 +494,80 @@ ServiceDispatchHandler.prototype.computePartialRange =
 function computePartialRange(serviceName, hostPort) {
     var self = this;
 
-    // Obtain and sort the affine worker and relay lists.
-    var relays = Object.keys(self.egressNodes.exitsFor(serviceName));
-    relays.sort();
     var serviceChannel = self.getOrCreateServiceChannel(serviceName);
-    var workers = serviceChannel.peers.keys();
-    workers.sort();
+
+    var range = {
+        relays: null,
+        workers: null,
+        relayIndex: NaN,
+        ratio: NaN,
+        start: NaN,
+        stop: NaN,
+        length: NaN,
+        affineWorkers: []
+    };
+
+    // Obtain and sort the affine worker and relay lists.
+    range.relays = Object.keys(self.egressNodes.exitsFor(serviceName));
+    range.relays.sort();
+    range.workers = serviceChannel.peers.keys();
```

Comment:
hey so... isn't this keys array shared on the peers collection? and then we just mutable sort it? and log it? and .... :rage: